aiida_quantumespresso/calculations/pp.py

```diff
@@ -1,27 +1,218 @@
+        elif dimension == 2:
+            parameters['PLOT']['output_format'] = 7    # Gnuplot, 2D
+        elif dimension == 3:
+            parameters['PLOT']['output_format'] = 6    # Gaussian cube, 3D
+        else:
+            raise exceptions.InputValidationError(
+                "Unsupported value, {}, for 'iflag'. Must be in the range 0 to 4".format(dimension)
+            )
+
+        namelists_toprint = self._default_namelists
+
+        input_filename = self.inputs.metadata.options.input_filename
+        with folder.open(input_filename, 'w') as infile:
+            for namelist_name in namelists_toprint:
+                infile.write(u'&{0}\n'.format(namelist_name))
+                # namelist content; set to {} if not present, so that we leave an empty namelist
+                namelist = parameters.pop(namelist_name, {})
+                for key, value in sorted(six.iteritems(namelist)):
+                    infile.write(convert_input_to_namelist_entry(key, value))
+                infile.write(u'/\n')
+
+
+        # Check for specified namelists that are not expected
+        if parameters:
+            raise exceptions.InputValidationError(
+                'The following namelists are specified in parameters, but are '
+                'not valid namelists for the current type of calculation: '
+                '{}'.format(','.join(list(parameters.keys()))))
+
+        remote_copy_list = []
+        local_copy_list = []
+
+        # Copy remote output dir
+        parent_calc_folder = self.inputs.get('parent_folder', None)
+        if isinstance(parent_calc_folder, orm.RemoteData):
+            remote_copy_list.append((
+                parent_calc_folder.computer.uuid,
+                os.path.join(parent_calc_folder.get_remote_path(), self._INPUT_SUBFOLDER),
+                self._OUTPUT_SUBFOLDER
+            ))
+            remote_copy_list.append((
+                parent_calc_folder.computer.uuid,
+                os.path.join(parent_calc_folder.get_remote_path(), self._PSEUDO_SUBFOLDER),
+                self._PSEUDO_SUBFOLDER
+            ))
+        elif isinstance(parent_calc_folder, orm.FolderData):
+            for filename in parent_calc_folder.list_object_names():
+                local_copy_list.append((
+                    parent_calc_folder.uuid,
+                    filename,
+                    os.path.join(self._OUTPUT_SUBFOLDER, filename)
+                ))
+                local_copy_list.append((
+                    parent_calc_folder.uuid,
+                    filename,
+                    os.path.join(self._PSEUDO_SUBFOLDER, filename)
+                ))
+
+        codeinfo = datastructures.CodeInfo()
+        codeinfo.stdin_name = self.inputs.metadata.options.input_filename
+        codeinfo.stdout_name = self.inputs.metadata.options.output_filename
+        codeinfo.code_uuid = self.inputs.code.uuid
+
+        calcinfo = datastructures.CalcInfo()
+        calcinfo.uuid = str(self.uuid)
```

Comment:
this line is not necessary. Is done automatically by AiiDA as of v1.0

Reply:
Just this singlular line?
`calcinfo.uuid = str(self.uuid)
`

Reply:
yes, since this always has to be done, I just moved the responsibility to `aiida-core`